pennylane/pytrees.py

```diff
@@ -30,6 +30,57 @@
 UnflattenFn = Callable[[Leaves, Metadata], Any]
 
 
+def flatten_list(obj: list):
+    """Flatten a list."""
+    return obj, None
+
+
+def flatten_tuple(obj: tuple):
+    """Flatten a tuple."""
+    return obj, None
+
+
+def flatten_dict(obj: dict):
+    """Flatten a dictionary."""
+    return obj.values(), tuple(obj.keys())
+
+
+flatten_registrations: Dict[type, FlattenFn] = {
```

Comment:
```suggestion
flatten_registrations: dict[type, FlattenFn] = {
```
Nitpick, but there's no need to use the builtin collection aliases since Pennylane no longer supports Python < 3.8 (correct me if this isn't the case)